### PR TITLE
Use Lodash's mergeWith and warn when losing reactivity

### LINKED ISSUE #589: Deltas: warn when we are working without reactivity
**Describe exactly what you would like to see in an upcoming release**

We had an issue in #586 where a value was not reactive, because it was not present in the `deltas.added.workflow.etc`, but appeared in `deltas.updated.workflow.etc`.

When the value was set, for Vue it was not reactive, as there was no `Observer` in the object.

When reading the docs for Lodash's `merge`, which we are using to merge deltas, I learned about `mergeWith`. The _with_ is because it receives a kind of callback function. When provided, every value merged will pass through this function, where we can customize the merge, skip it, etc.

An idea that occurred to me was to merge as we are doing now. But also check whether the property exists (we need to do something like `if isObject(obj)`). If it doesn't, then we could `console.warn` to let users/devs know about a value that was added without reactivity.

That should never occur… but better safe than sorry, and it should be a cheap verification.

**Additional context**

**Pull requests welcome!**

### PR DESCRIPTION
**Pull request by kinow**
These changes close #589 (and removes an unnecessary `console.log` from tests)

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Appropriate tests are included (unit and/or functional).
- [x] No change log entry required (why? e.g. invisible to users).
- [x] No documentation update required.

### CODE REVIEW

**Comment by kinow** [comment]
With this change, and without #586 , when we hold or release a workflow we get

image

**Comment by kinow** [comment]
I think this should be ready for review. Just waiting for CI to complete, and codecov to confirm the code is covered.

@hjoliver after this change, we should see a warning, until #586 is merged. After #586 is merged, we shouldn't — in theory — see any warning again. But if we see the warning in the console, we won't have to spend time investigating reactivity issues in deltas (no fun!) :+1:

**Comment by kinow** [comment]
Funny. The macos build failed. Re-ran all jobs, then now ubuntu+firefox failed. :confused: 

image

Working for me locally. Third time's the charm. Let's kick it once again.

**Comment by oliver-sanders** [comment]
Perhaps we have a flaky test, which test was failing?

**Comment by kinow** [comment]
> Perhaps we have a flaky test, which test was failing?

The `build` workflow error was due to a `no-console` that I forgot. The `macos` build (or job, sorry I keep using both with GH actions) keeps failing when it tries to execute Cypress, before running the tests.

Today: https://github.com/cylc/cylc-ui/actions/runs/555846220

```
 info There appears to be trouble with your network connection. Retrying...
info There appears to be trouble with your network connection. Retrying...
info There appears to be trouble with your network connection. Retrying...
info There appears to be trouble with your network connection. Retrying...
error An unexpected error occurred: "https://registry.yarnpkg.com/rxjs/-/rxjs-6.6.2.tgz: ESOCKETTIMEDOUT".
info If you think this is a bug, please open a bug report with the information provided in "/Users/runner/work/cylc-ui/cylc-ui/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
info There appears to be trouble with your network connection. Retrying...
Error: The process '/Users/runner/.yarn/bin/yarn' failed with exit code 1
```

From 2 days ago, this branch too: https://github.com/cylc/cylc-ui/actions/runs/552663688

```
Done in 266.80s.
/usr/local/bin/npx cypress cache list
┌─────────┬──────────────┐
│ version │ last used    │
├─────────┼──────────────┤
│ 6.1.0   │ 2 months ago │
└─────────┴──────────────┘
/usr/local/bin/npx cypress verify

25l[21:33:50]  Verifying Cypress can run /Users/runner/.cache/Cypress/6.1.0/Cypress.app [started]
[21:34:20]  Verifying Cypress can run /Users/runner/.cache/Cypress/6.1.0/Cypress.app [failed]
25hCypress verification timed out.

This command failed with the following output:

/Users/runner/.cache/Cypress/6.1.0/Cypress.app/Contents/MacOS/Cypress --no-sandbox --smoke-test --ping=107

----------

Command timed out after 30000 milliseconds: /Users/runner/.cache/Cypress/6.1.0/Cypress.app/Contents/MacOS/Cypress --no-sandbox --smoke-test --ping=107
Timed out

----------

Platform: darwin (19.6.0)
Cypress Version: 6.1.0
25h::debug::The process '/usr/local/bin/npx' failed with exit code 1
Error: The process '/usr/local/bin/npx' failed with exit code 1
```

**Comment by oliver-sanders** [comment]
Ok, have an idea what might be causing that, https://github.com/cylc/cylc-ui/pull/592

**Comment by oliver-sanders** [comment]
Tried upgrading to Yarn2, however, got a load of errors, will need to go through the migration guide properly. Perhaps worth waiting for Yarn2 to stabilise first, dumping in an issue for now.

https://github.com/cylc/cylc-ui/issues/593

**Review by oliver-sanders — APPROVED** [review]

**Comment by oliver-sanders** [comment]
Test failure nothing to do with this branch.

**Comment by kinow** [comment]
I think I found an alternative to this PR, that would add reactivity to an object, instead of just warning. Testing will take a little bit (browser + IDE)

**Comment by kinow** [comment]
I'm pushing an improved solution to #530 as that branches also has issues with reactivity after deltas are merged. Once that one is merged, will leave this one open a little longer while I look at other PR's, to make sure this issue is really gone.

**Comment by kinow** [comment]
So far looks like the new solution worked on #530 . If that PR is closed, then this one can be closed as superseded (it has the same code, with a small modification in the merge customizer using `Vue.set` instead of just warning).

**Comment by kinow** [comment]
Confirmed it's closed by #530

**Pull request closed without merging**
